homeassistant/components/weatherflow/config_flow.py

```diff
@@ -0,0 +1,101 @@
+"""Config flow for WeatherFlow."""
+from __future__ import annotations
+
+import asyncio
+from typing import Any
+
+from async_timeout import timeout
+
+# import my_pypi_dependency
+from pyweatherflowudp.client import EVENT_DEVICE_DISCOVERED, WeatherFlowListener
+from pyweatherflowudp.const import DEFAULT_HOST
+from pyweatherflowudp.errors import AddressInUseError, ListenerError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import config_entry_flow
+
+from .const import DOMAIN, LOGGER
+
+STEP_USER_DATA_SCHEMA = vol.Schema({vol.Required(CONF_HOST, default=DEFAULT_HOST): str})
+
+
+async def _async_has_devices(hass: HomeAssistant) -> bool:
+    """Return if there are devices that can be discovered."""
+
+    event = asyncio.Event()
+    host: str = DEFAULT_HOST
+
+    @callback
+    def found():
+        """Handle a discovered device."""
+        event.set()
+
+    async with WeatherFlowListener(host) as client:
+        LOGGER.info("Registering EVENT_DISCOVERED_FUNCTION")
+        client.on(EVENT_DEVICE_DISCOVERED, lambda _: found())
+        try:
+            async with timeout(10):
+                await event.wait()
+        except asyncio.TimeoutError:
+            return False
+    return True
+
+
+config_entry_flow.register_discovery_flow(DOMAIN, "WeatherFlow", _async_has_devices)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for WeatherFlow."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        current_hosts = [
```

Comment:
Can use `self._async_abort_entries_match(user_input)` to test if existing already here

Reply:
I think this is fixed now.